spec/system/user_reset_progress_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'rails_helper'
+
+RSpec.describe 'User Reset Progress', type: :system do
+  let!(:foundations_path) { create(:path, title: 'Foundations', default_path: true) }
+  let!(:foundation_course) { create(:course, title: 'Foundations', path: foundations_path) }
+  let!(:foundation_section) { create(:section, course: foundation_course) }
+  let!(:foundation_first_lesson) { create(:lesson, title: 'First Foundation Lesson', section: foundation_section) }
+  let!(:foundation_second_lesson) { create(:lesson, section: foundation_section) }
+  let!(:rails_path) { create(:path, title: 'Rails') }
+  let!(:rails_course) { create(:course, title: 'Rails', path: rails_path) }
+  let!(:rails_section) { create(:section, course: rails_course) }
+  let!(:rails_first_lesson) { create(:lesson, title: 'First Rails Lesson', section: rails_section) }
+  let!(:rails_second_lesson) { create(:lesson, section: rails_section) }
+  let!(:user) { create(:user) }
+
+  it 'deletes all lesson completions and resets to default path' do
+    sign_in(user)
+    visit dashboard_path
+
+    within(find('.skills')) do
+      expect(page).to have_content(foundation_course.title)
+      expect(find(:test_id, 'default-badge')).to have_text('')
+      expect(find(:test_id, 'foundations-start-btn')).to have_text('Start')
+    end
+
+    visit path_course_lesson_path(foundations_path, foundation_course, foundation_first_lesson)
+    within(find('.lesson-button-group')) do
```

Comment:
Instead of completing the lessons through the UI, what do you think about creating some lesson completion records before the the test is run instead?

My thinking is we already have that behaviour tested elsewhere so its redundant here and would simplify this test if we didn't have to manually complete lessons.

Reply:
Oh, yes, I really like this idea and will re-work this system test.